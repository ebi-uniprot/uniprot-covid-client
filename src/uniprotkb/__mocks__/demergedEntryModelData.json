{
  "entryType": "Inactive",
  "primaryAccession": "Q00007",
  "uniProtId": "2ABA_HUMAN",
  "annotationScore": 0.0,
  "inactiveReason": {
    "inactiveReasonType": "DEMERGED",
    "mergeDemergeTo": ["P63151", "P63150"]
  }
}