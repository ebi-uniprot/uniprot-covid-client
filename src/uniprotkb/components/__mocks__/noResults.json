{
  "facets": [{
    "label": "Status",
    "name": "reviewed",
    "allowMultipleSelection": false
  }, {
    "label": "Popular organisms",
    "name": "popular_organism",
    "allowMultipleSelection": true
  }, {
    "label": "Proteins with",
    "name": "proteins_with",
    "allowMultipleSelection": true
  }, {
    "label": "Protein Existence",
    "name": "existence",
    "allowMultipleSelection": true
  }, {
    "label": "Annotation Score",
    "name": "annotation_score",
    "allowMultipleSelection": true
  }, {
    "label": "Sequence length",
    "name": "length",
    "allowMultipleSelection": true,
    "values": [{
      "label": "1 - 200",
      "value": "[1 TO 200]",
      "count": 0
    }, {
      "label": "201 - 400",
      "value": "[201 TO 400]",
      "count": 0
    }, {
      "label": "401 - 600",
      "value": "[401 TO 600]",
      "count": 0
    }, {
      "label": "601 - 800",
      "value": "[601 TO 800]",
      "count": 0
    }, {
      "label": ">= 801",
      "value": "[801 TO *]",
      "count": 0
    }]
  }],
  "matchedFields": [],
  "results": []
}