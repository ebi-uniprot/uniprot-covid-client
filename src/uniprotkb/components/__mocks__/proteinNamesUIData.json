{
  "recommendedName": "My protein",
  "shortNames": "some shorter names",
  "alternativeNames": ["alt1", "alt2"]
}
