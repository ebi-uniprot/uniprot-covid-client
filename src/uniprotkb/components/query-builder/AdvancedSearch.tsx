import React, { Component, FormEvent, MouseEvent } from 'react';
import ClauseList from './ClauseList';
import {
  Namespace,
  Evidence,
  Clause,
  SearchTermType,
  Operator,
  Evidences,
} from '../../types/searchTypes';
import './styles/advanced-search.scss';

type AdvancedSearchProps = {
  searchTerms: SearchTermType[];
  queryString: string;
  namespace: Namespace;
  clauses: Clause[];
  evidences: Evidences;
  dispatchfetchEvidencesIfNeeded: (type: Evidence) => void;
  dispatchFetchSearchTermsIfNeeded: () => void;
  handleAdvancedSubmitClick: (event: FormEvent | MouseEvent) => void;
  dispatchAddClause: () => void;
  handleFieldSelect: (clauseId: string, field: SearchTermType) => void;
  handleInputChange: (clauseId: string, value: string, id?: string) => void;
  handleEvidenceChange: (clauseId: string, value: string) => void;
  handleRangeInputChange: (
    clauseId: string,
    value: string,
    from?: boolean
  ) => void;
  handleLogicChange: (clauseId: string, value: Operator) => void;
  handleRemoveClause: (clauseId: string) => void;
  dispatchSetPreSelectedClauses: () => void;
};
class AdvancedSearch extends Component<AdvancedSearchProps> {
  componentDidMount() {
    const {
      dispatchfetchEvidencesIfNeeded,
      dispatchFetchSearchTermsIfNeeded,
    } = this.props;
    dispatchfetchEvidencesIfNeeded(Evidence.GO);
    dispatchfetchEvidencesIfNeeded(Evidence.ANNOTATION);
    dispatchFetchSearchTermsIfNeeded();
  }

  render() {
    const {
      handleAdvancedSubmitClick,
      namespace,
      dispatchAddClause,
    } = this.props;
    return (
      <form
        className="advanced-search"
        onSubmit={handleAdvancedSubmitClick}
        data-testid="advanced-search-form"
      >
        <fieldset>
          <label htmlFor="namespace-select">
            Searching in
            <select id="namespace-select">
              <option>{namespace}</option>
            </select>
          </label>
        </fieldset>
        <fieldset>
          <ClauseList {...this.props} />
        </fieldset>
        <div className="advanced-search__actions">
          <button
            type="button"
            id="add-field"
            className="button tertiary"
            data-testid="advanced-search-add-field"
            onClick={dispatchAddClause}
          >
            Add Field
          </button>
          <button type="submit" id="submit-query" className="button">
            Search
          </button>
        </div>
      </form>
    );
  }
}

export default AdvancedSearch;
