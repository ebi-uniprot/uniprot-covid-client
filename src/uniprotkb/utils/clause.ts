import { v1 } from 'uuid';
import { isEqual } from 'lodash-es';
import { removeProperty } from '../../shared/utils/utils';
import { Operator, Clause, itemType, dataType } from '../types/searchTypes';

export const createEmptyClause = (): Clause => ({
  id: v1(),
  logicOperator: Operator.AND,
  searchTerm: {
    label: 'All',
    term: 'All',
    example: 'a4_human, P05067, cdc7 human',
    itemType: itemType.single,
    dataType: dataType.string,
    id: 'id_all',
  },
  queryInput: {},
});

export const createPreSelectedClauses = (): Clause[] => [
  {
    id: v1(),
    logicOperator: Operator.AND,
    searchTerm: {
      label: 'Gene Name [GN]',
      term: 'gene',
      example: 'ydj1',
      itemType: itemType.single,
      dataType: dataType.string,
      id: 'id_gene',
    },
    queryInput: {},
  },
  {
    id: v1(),
    logicOperator: Operator.AND,
    searchTerm: {
      id: 'id_taxonomy',
      label: 'Taxonomy [OC]',
      term: 'taxonomy',
      itemType: itemType.single,
      dataType: dataType.string,
      autoComplete: '/uniprot/api/suggester?dict=taxonomy&query=?',
      description: 'Search by NCBI taxonomy',
      example: 'human',
      termSuffix: true,
    },
    queryInput: {},
  },
  {
    id: v1(),
    logicOperator: Operator.AND,
    searchTerm: {
      id: 'id_reviewed',
      label: 'Reviewed',
      term: 'reviewed',
      example: 'true',
      itemType: itemType.single,
      dataType: dataType.enum,
      values: [
        { name: 'Yes', value: 'true' },
        { name: 'No', value: 'false' },
      ],
    },
    queryInput: {},
  },
  {
    id: v1(),
    logicOperator: Operator.AND,
    searchTerm: {
      id: 'id_keyword',
      label: 'Keyword [KW]',
      term: 'keyword',
      example: 'chromosomal',
      itemType: itemType.single,
      dataType: dataType.string,
      description: 'Search by keyword',
      autoComplete: '/uniprot/api/suggester?dict=keyword&query=?',
    },
    queryInput: {},
  },
  createEmptyClause(),
];

export const clausesAreEqual = (clause1: Clause, clause2: Clause) =>
  isEqual(removeProperty(clause1, 'id'), removeProperty(clause2, 'id'));
