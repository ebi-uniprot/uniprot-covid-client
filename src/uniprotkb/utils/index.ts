import { uniq } from 'lodash-es';
import UniProtKBEntryConfig from '../config/UniProtEntryConfig';
import { UniProtkbUIModel } from '../adapters/uniProtkbConverter';
import { GeneNamesData } from '../adapters/namesAndTaxonomyConverter';
import { Property, PropertyKey } from '../types/modelTypes';

/* eslint-disable @typescript-eslint/no-explicit-any */
export const hasContent = (obj: any) => {
  return Object.values(obj).some((val) => {
    if (Array.isArray(val)) {
      const valArray = val as any[];
      return valArray.length > 0;
    }
    if (typeof val === 'object' && val) {
      if (val instanceof Map) {
        return Array.from(val.values()).some(
          (value) => value && value.length > 0
        );
      }
      return Object.values(val).length > 0;
    }
    return false;
  });
};

export const hasExternalLinks = (transformedData: UniProtkbUIModel) =>
  UniProtKBEntryConfig.some(({ name }) => {
    const data = transformedData[name];
    return typeof data.xrefData && data.xrefData.length > 0;
  });

export const flattenGeneNameData = (geneNamesData: GeneNamesData) => {
  const geneNames: string[] = [];
  geneNamesData.forEach(
    ({ geneName, synonyms = [], orfNames = [], orderedLocusNames = [] }) => {
      if (geneName) {
        geneNames.push(geneName.value);
      }
      [synonyms, orfNames, orderedLocusNames].forEach((names) => {
        names.forEach(({ value }) => {
          geneNames.push(value);
        });
      });
    }
  );
  return uniq(geneNames);
};

export const transfromProperties = (properties: Property[]) => {
  const o: { [key: string]: string } = {};
  properties.forEach(({ key, value }) => {
    if (key && value) {
      o[key] = value;
    }
  });
  return o;
};

// This function is useful because our API returns arrays of objects of shape: { key: x, value: y}
export const getPropertyValue = (
  properties: Property[],
  propertyKey: PropertyKey
) => {
  const found = properties.find(({ key }) => key === propertyKey);
  return found ? found.value : null;
};

// The regex that matches uniprot accession. Taken from:
// https://www.uniprot.org/help/accession_numbers
export const uniProtKBAccessionRegEx = /[OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9]([A-Z][A-Z0-9]{2}[0-9]){1,2}/i;
