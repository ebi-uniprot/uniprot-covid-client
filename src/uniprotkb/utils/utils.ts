import { uniq } from 'lodash-es';
import UniProtKBEntryConfig from '../config/UniProtEntryConfig';
import { UniProtkbUIModel } from '../adapters/uniProtkbConverter';
import { GeneNamesData } from '../adapters/namesAndTaxonomyConverter';
import { Property, PropertyKey } from '../types/modelTypes';

/* eslint-disable @typescript-eslint/no-explicit-any */
export const hasContent = (obj: any) => {
  return Object.values(obj).some((val) => {
    if (Array.isArray(val)) {
      const valArray = val as any[];
      return valArray.length > 0;
    }
    if (typeof val === 'object' && val) {
      if (val instanceof Map) {
        return Array.from(val.values()).some(
          (value) => value && value.length > 0
        );
      }
      return Object.values(val).length > 0;
    }
    return false;
  });
};

export const hasExternalLinks = (transformedData: UniProtkbUIModel) =>
  UniProtKBEntryConfig.some(({ name }) => {
    const data = transformedData[name];
    return typeof data.xrefData && data.xrefData.length > 0;
  });

export const flattenGeneNameData = (geneNamesData: GeneNamesData) => {
  const geneNames: string[] = [];
  geneNamesData.forEach(
    ({ geneName, synonyms = [], orfNames = [], orderedLocusNames = [] }) => {
      if (geneName) {
        geneNames.push(geneName.value);
      }
      [synonyms, orfNames, orderedLocusNames].forEach((names) => {
        names.forEach(({ value }) => {
          geneNames.push(value);
        });
      });
    }
  );
  return uniq(geneNames);
};

export const transfromProperties = (properties: Property[]) => {
  const o: { [key: string]: string } = {};
  properties.forEach(({ key, value }) => {
    if (key && value) {
      o[key] = value;
    }
  });
  return o;
};

// This function is useful because our API returns arrays of objects of shape: { key: x, value: y}
export const getPropertyValue = (
  properties: Property[],
  propertyKey: PropertyKey
) => {
  const found = properties.find(({ key }) => key === propertyKey);
  return found ? found.value : null;
};
