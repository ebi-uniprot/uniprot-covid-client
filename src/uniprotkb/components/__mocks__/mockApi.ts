import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import results from './results.json';
import noResults from './noResults.json';
import entry from './swissprotEntry.json';
import resultFields from '../../__mocks__/resultFields.json';
import mockFasta from './fasta.json';

const mock = new MockAdapter(axios);
mock.onGet(/.+noresult/).reply(200, noResults, { 'x-total-records': 25 });
mock
  .onGet(/\/uniprotkb\/search/)
  .reply(200, results, { 'x-total-records': 25 });
mock.onGet(/\/uniprotkb\/result-fields/).reply(200, resultFields);
mock
  .onGet(
    /\/uniprotkb\/download\?compressed=true&format=fasta&query=nod2&size=10/
  )
  .reply(200, mockFasta, { 'content-type': 'text/fasta' });
mock.onGet(/\/uniprotkb\//).reply(200, entry);

export default mock;
