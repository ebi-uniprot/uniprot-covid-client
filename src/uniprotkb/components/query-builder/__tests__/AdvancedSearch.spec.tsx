import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { createEmptyClause } from '../../../utils/clause';
import AdvancedSearch from '../AdvancedSearch';
import { resetUuidV1 } from '../../../../../__mocks__/uuid';

let rendered;
let props;

describe('AdvancedSearch shallow components', () => {
  beforeEach(() => {
    resetUuidV1();
    props = {
      dispatchAddClause: jest.fn(),
      handleAdvancedSubmitClick: jest.fn(),
      dispatchFetchSearchTermsIfNeeded: jest.fn(),
      dispatchfetchEvidencesIfNeeded: jest.fn(),
      dispatchCopyQueryClausesToSearch: jest.fn(),
      dispatchSetPreSelectedClauses: jest.fn(),
      history: {
        push: jest.fn(),
      },
      clauses: [...Array(4)].map(() => createEmptyClause()),
      namespace: 'UniProtKB',
      searchTerms: [],
      evidences: {
        go: {
          data: [],
          isFetching: false,
        },
        annotation: {
          data: [],
          isFetching: false,
        },
      },
    };
    rendered = render(<AdvancedSearch {...props} />);
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should call to fetch search terms', () => {
    expect(props.dispatchFetchSearchTermsIfNeeded).toHaveBeenCalled();
  });

  test('should call to fetch evidences', () => {
    expect(props.dispatchfetchEvidencesIfNeeded).toHaveBeenCalledWith('go');
    expect(props.dispatchfetchEvidencesIfNeeded).toHaveBeenCalledWith(
      'annotation'
    );
  });

  test('should add field rows', () => {
    const { getByTestId } = rendered;
    fireEvent.click(getByTestId('advanced-search-add-field'));
    expect(props.dispatchAddClause).toHaveBeenCalled();
  });

  test('should submit a query', () => {
    const { getByTestId } = rendered;
    fireEvent.submit(getByTestId('advanced-search-form'));
    expect(props.handleAdvancedSubmitClick).toHaveBeenCalled();
  });
});
