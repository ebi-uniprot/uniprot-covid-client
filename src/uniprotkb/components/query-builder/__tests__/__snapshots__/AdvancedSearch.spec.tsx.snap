// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedSearch shallow components should render 1`] = `
<DocumentFragment>
  <form
    class="advanced-search"
    data-testid="advanced-search-form"
  >
    <fieldset>
      <label
        for="namespace-select"
      >
        Searching in
        <select
          id="namespace-select"
        >
          <option>
            UniProtKB
          </option>
        </select>
      </label>
    </fieldset>
    <fieldset>
      <div
        class="advanced-search__clause"
      >
        <select
          class="advanced-search__logic"
          data-testid="advanced-search-logic-select"
        >
          <option
            value="AND"
          >
            AND
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="NOT"
          >
            NOT
          </option>
        </select>
        <div
          class="dropdown-container"
        >
          <button
            class="button dropdown "
            type="button"
          >
            All
          </button>
          <div
            class="dropdown-menu"
          >
            <div
              class="autocomplete-container"
            >
              <span
                class="search-input"
              >
                <input
                  data-testid="search-input"
                  placeholder="Search for field"
                  type="text"
                  value=""
                />
                <span
                  class="search-input__suffix"
                  data-testid="search-input-suffix"
                  role="presentation"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 512 512"
                    width="14"
                  >
                    <path
                      d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
                    />
                  </svg>
                </span>
              </span>
              <div
                class="autocomplete-menu"
              >
                <div
                  class="dropdown-menu__panel"
                />
              </div>
            </div>
            <div
              class="dropdown-menu__panel"
            >
              <ul
                class=""
              />
            </div>
          </div>
        </div>
        <div
          class="advanced-search__inputs"
        >
          <div
            class="advanced-search__inputs"
          >
            <label
              for="input_All"
            >
              All
              <input
                id="input_All"
                placeholder="a4_human, P05067, cdc7 human"
                type="text"
                value=""
              />
            </label>
          </div>
        </div>
        <button
          class="button tertiary button-remove"
          data-testid="clause-list-button-remove"
          type="button"
        >
          Remove
        </button>
      </div>
      <div
        class="advanced-search__clause"
      >
        <select
          class="advanced-search__logic"
          data-testid="advanced-search-logic-select"
        >
          <option
            value="AND"
          >
            AND
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="NOT"
          >
            NOT
          </option>
        </select>
        <div
          class="dropdown-container"
        >
          <button
            class="button dropdown "
            type="button"
          >
            All
          </button>
          <div
            class="dropdown-menu"
          >
            <div
              class="autocomplete-container"
            >
              <span
                class="search-input"
              >
                <input
                  data-testid="search-input"
                  placeholder="Search for field"
                  type="text"
                  value=""
                />
                <span
                  class="search-input__suffix"
                  data-testid="search-input-suffix"
                  role="presentation"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 512 512"
                    width="14"
                  >
                    <path
                      d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
                    />
                  </svg>
                </span>
              </span>
              <div
                class="autocomplete-menu"
              >
                <div
                  class="dropdown-menu__panel"
                />
              </div>
            </div>
            <div
              class="dropdown-menu__panel"
            >
              <ul
                class=""
              />
            </div>
          </div>
        </div>
        <div
          class="advanced-search__inputs"
        >
          <div
            class="advanced-search__inputs"
          >
            <label
              for="input_All"
            >
              All
              <input
                id="input_All"
                placeholder="a4_human, P05067, cdc7 human"
                type="text"
                value=""
              />
            </label>
          </div>
        </div>
        <button
          class="button tertiary button-remove"
          data-testid="clause-list-button-remove"
          type="button"
        >
          Remove
        </button>
      </div>
      <div
        class="advanced-search__clause"
      >
        <select
          class="advanced-search__logic"
          data-testid="advanced-search-logic-select"
        >
          <option
            value="AND"
          >
            AND
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="NOT"
          >
            NOT
          </option>
        </select>
        <div
          class="dropdown-container"
        >
          <button
            class="button dropdown "
            type="button"
          >
            All
          </button>
          <div
            class="dropdown-menu"
          >
            <div
              class="autocomplete-container"
            >
              <span
                class="search-input"
              >
                <input
                  data-testid="search-input"
                  placeholder="Search for field"
                  type="text"
                  value=""
                />
                <span
                  class="search-input__suffix"
                  data-testid="search-input-suffix"
                  role="presentation"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 512 512"
                    width="14"
                  >
                    <path
                      d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
                    />
                  </svg>
                </span>
              </span>
              <div
                class="autocomplete-menu"
              >
                <div
                  class="dropdown-menu__panel"
                />
              </div>
            </div>
            <div
              class="dropdown-menu__panel"
            >
              <ul
                class=""
              />
            </div>
          </div>
        </div>
        <div
          class="advanced-search__inputs"
        >
          <div
            class="advanced-search__inputs"
          >
            <label
              for="input_All"
            >
              All
              <input
                id="input_All"
                placeholder="a4_human, P05067, cdc7 human"
                type="text"
                value=""
              />
            </label>
          </div>
        </div>
        <button
          class="button tertiary button-remove"
          data-testid="clause-list-button-remove"
          type="button"
        >
          Remove
        </button>
      </div>
      <div
        class="advanced-search__clause"
      >
        <select
          class="advanced-search__logic"
          data-testid="advanced-search-logic-select"
        >
          <option
            value="AND"
          >
            AND
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="NOT"
          >
            NOT
          </option>
        </select>
        <div
          class="dropdown-container"
        >
          <button
            class="button dropdown "
            type="button"
          >
            All
          </button>
          <div
            class="dropdown-menu"
          >
            <div
              class="autocomplete-container"
            >
              <span
                class="search-input"
              >
                <input
                  data-testid="search-input"
                  placeholder="Search for field"
                  type="text"
                  value=""
                />
                <span
                  class="search-input__suffix"
                  data-testid="search-input-suffix"
                  role="presentation"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 512 512"
                    width="14"
                  >
                    <path
                      d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
                    />
                  </svg>
                </span>
              </span>
              <div
                class="autocomplete-menu"
              >
                <div
                  class="dropdown-menu__panel"
                />
              </div>
            </div>
            <div
              class="dropdown-menu__panel"
            >
              <ul
                class=""
              />
            </div>
          </div>
        </div>
        <div
          class="advanced-search__inputs"
        >
          <div
            class="advanced-search__inputs"
          >
            <label
              for="input_All"
            >
              All
              <input
                id="input_All"
                placeholder="a4_human, P05067, cdc7 human"
                type="text"
                value=""
              />
            </label>
          </div>
        </div>
        <button
          class="button tertiary button-remove"
          data-testid="clause-list-button-remove"
          type="button"
        >
          Remove
        </button>
      </div>
    </fieldset>
    <div
      class="advanced-search__actions"
    >
      <button
        class="button tertiary"
        data-testid="advanced-search-add-field"
        id="add-field"
        type="button"
      >
        Add Field
      </button>
      <button
        class="button"
        id="submit-query"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</DocumentFragment>
`;
